src/interpreter.rs

```diff
@@ -423,15 +444,92 @@ impl<'a> Interpreter<'a> {
         Ok(acc)
     }
 
-    fn evaluate_primary<W: Write>(&self, primary: &Primary, stderr: &mut W) -> Result<f64, Error> {
+    fn evaluate_primary<W: Write>(
+        &mut self,
```

Comment:
Is `self` mutable because of the random number generator?

Reply:
Yeah, exactly. I thought about using inferior mutability for that later. Actually, the idea would be to create a devices struct which would contain all devices like rng, stdout, stdin, screen, etc... and only this one will need to be mut.